packages/rest/src/controllers/credentials/CredentialController.ts

```diff
@@ -31,8 +31,17 @@ export class CredentialController extends Controller {
    */
   @Example<CredentialExchangeRecordProps[]>([CredentialExchangeRecordExample])
   @Get('/')
-  public async getAllCredentials() {
-    const credentials = await this.agent.credentials.getAll()
+  public async getAllCredentials(
+    @Query('threadId') threadId?: string,
+    @Query('connectionId') connectionId?: string,
+    @Query('state') state?: string,
+  ) {
+    let credentials = await this.agent.credentials.getAll()
+
+    if (state) credentials = credentials.filter((c) => c.state === state)
+    if (threadId) credentials = credentials.filter((c) => c.threadId === threadId)
```

Comment:
Ah I see this is also done like this in the connections controller, but we should actually filter in the query, to not retrieve records that aren't needed.

We should expose this in AFJ (I'll create an issue for this), but for now we can do it like this:

```ts
    const credentialRepository = this.agent.dependencyManager.resolve(CredentialRepository)

    const credentials = await credentialRepository.findByQuery({
      connectionId,
      threadId,
      state,
    })

    return credentials.map((c) => c.toJSON())
```

When this is merged, could you also make a PR to update this for the connections controller?

@janrtvld FYI

Reply:
https://github.com/hyperledger/aries-framework-javascript/issues/1022